Sources/SimpleCurrencyConverter/SimpleCurrencyConverter.swift

```diff
@@ -1,6 +1,135 @@
+            self.exchangeRatesAPIKey = exchangeRatesAPIKey
+        } else {
+            self.exchangeRatesAPIKey = getAPIKeyFromPlistFile()
+        }
+
+        if let urlSession = urlSession {
+            self.networkManager = ExchangeRatesNetworkManager(session: urlSession)
+        } else {
+            self.networkManager = ExchangeRatesNetworkManager(session: URLSession.shared)
+        }
+    }
+    
+    public func getExchangeRate(baseCurrency: Currency, targetCurrencies: [Currency],
+                         resolve: @escaping ([String : Double]) -> Void, reject: @escaping (Error) -> Void) {
+        if #available(iOS 13.0, *) {
+            Task {
+                let result = try await networkManager?.requestExchangeRate(
+                    apiKey: exchangeRatesAPIKey, url: getExhangeRatesURL().get(), base: baseCurrency, target: targetCurrencies)
+                switch result {
+                case .success(let rates):
+                    resolve(rates)
+                case .failure(let error):
+                    reject(error)
+                default: ()
+                }
+            }
+        }
     }
+    
+    public func convert(amount: Double, baseCurrency: Currency, targetCurrency: Currency,
+                 resolve: @escaping (Double) -> Void, reject: @escaping (Error) -> Void) {
+        if #available(iOS 13.0, *) {
+            Task {
+                let result = try await networkManager?.requestConvert(
+                    apiKey: exchangeRatesAPIKey, url: getConvertURL().get(), amount: amount, base: baseCurrency, target: targetCurrency)
+                switch result {
+                case .success(let result):
+                    resolve(result)
+                case .failure(let error):
+                    reject(error)
+                default: ()
+                }
+            }
+        }
+    }
+    
+    public func convert(amount: Double, baseCurrency: Currency, targetCurrencies: [Currency],
+                 resolve: @escaping ([String : Double]) -> Void, reject: @escaping (Error) -> Void) {
+        if #available(iOS 13.0, *) {
+            Task {
+                let result = try await networkManager?.requestExchangeRate(
+                    apiKey: exchangeRatesAPIKey, url: getExhangeRatesURL().get(), base: baseCurrency, target: targetCurrencies)
+                switch result {
+                case .success(let rates):
+                    var modifiedRates = rates
+                    for rate in modifiedRates {
+                        let value = rate.value * amount
+                        modifiedRates.updateValue(value, forKey: rate.key)
+                    }
+                    resolve(modifiedRates)
+                case .failure(let error):
+                    reject(error)
+                default: ()
+                }
+            }
+        }
+    }
+}
+
+private extension SimpleCurrencyConverter {
+    
+    func getAPIKeyFromPlistFile() -> String {
```

Comment:
Can you remove get prefix and can you computed property instead?
```
var aPIKeyFromPlistFile: String {...}
```
Same comment for `getExhangeRatesURL` and `getConvertURL`.